glueforward/slskd.py

```
@@ -0,0 +1,140 @@
+            raise SlskdUnreachable(self.__client.base_url) from exception
+        except httpx.HTTPStatusError as exception:
+            self._handle_request_exception(
+                exception,
+                SlskdAuthFailed,
+                SlskdSetPortFailed
+            )
+            return None
+
+    def authenticate(self) -> None:
+        if self.get_is_authenticated():
+            return
+        logging.debug("Authenticating to slskd")
+        response = self.__request(
+            method="post",
+            url="/api/v0/session",
+            json=self.__credentials
+        )
+        self.__token = response.json().get('token')
+        logging.debug("slskd client authenticated")
+
+    def reset_authentication(self) -> None:
+        self.__token = None
+        logging.debug("slskd client authentication reset")
+
+    def __get_current_config(self) -> dict:
+        """Get current YAML config from slskd"""
+        response = self.__request(method="get", url="/api/v0/options/yaml")
+        try:
+            response_json = response.json()
+            if isinstance(response_json, str):
+                return yaml.load(response_json) or {}
+            return yaml.load(response_json.get('yaml', '')) or {}
+        except ValueError:
+            return yaml.load(response.text) or {}
+
+    def __update_config(self, config: dict) -> None:
+        """Update slskd config with new YAML"""
+        stream = StringIO()
+        yaml.dump(config, stream)
+        self.__request(
+            method="post",
+            url="/api/v0/options/yaml",
+            content=json.dumps(stream.getvalue()),
+            headers={"Content-Type": "application/json"}
+        )
+
+    def set_port(self, port: int) -> None:
+        """Update the listening port in slskd config via API"""
+        if not 1024 <= port <= 65535:
+            logging.warning("Invalid port %d, defaulting to 50300", port)
+            port = 50300
```

In my opinion, we should raise a retryable error instead, if what we wanted to do is not possible, then do nothing (no override with the default).

Something like `class SlskdIllegalPort(RetryableGlueforwardError)` would work.
Unfortunately, we cannot trigger a port reassignment on Gluetun, so we'll have to wait until a new port is assigned. I'm not really fond of that compromise, but that's the best we can do for now...

---

Agreed, it's probably better to leave the config as-is instead of updating it. Added the `SlskdIllegalPort` class for this.